CI passes on failed tests

`test_goal_apply_tactic` is failing on all the CI runs, but it is still marked as passed.
```text
failures:

---- test_goal_apply_tactic stdout ----
thread 'test_goal_apply_tactic' panicked at 'called `Result::unwrap()` on an `Err` value: "operator (=> a b) not supported, apply simplifier before invoking translator"', z3/tests/lib.rs:1230:42
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test_goal_apply_tactic

test result: FAILED. 64 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.87s

error: test failed, to rerun pass '-p z3 --test lib'
Error: Process completed with exit code 101.
```

Also `test_solver_unknown` seems also be causing intermittent test failures. Z3Prover/z3#5702

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -6,28 +6,31 @@ on:
   pull_request:
     branches: [ master ]
 
+env:
+  CARGO_INCREMENTAL: 0
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - name: Install Z3
-      run: sudo apt install libz3-dev
+      run: sudo apt-get install libz3-dev
     - name: Build
       run: cargo build -vv --all
       # XXX: Ubuntu's Z3 package seems to be missing some symbols, like
       # `Z3_mk_pbeq`, leading to linker errors. Just ignore this, I guess, until
       # we figure out how to work around it. At least we have the
       # statically-linked Z3 tests below...
-      continue-on-error: true
+      if: ${{ success() || failure() }}
     - name: Run tests
       run: cargo test -vv --all
       # See above.
-      continue-on-error: true
+      if: ${{ success() || failure() }}
     - name: Run tests with `arbitrary-size-numeral` enabled
       run: cargo test --manifest-path z3/Cargo.toml -vv --features arbitrary-size-numeral
       # See above.
-      continue-on-error: true
+      if: ${{ success() || failure() }}
 
   build_z3_statically:
     strategy:

--- a/z3/tests/lib.rs
+++ b/z3/tests/lib.rs
@@ -580,6 +580,7 @@ fn test_rec_func_def_unsat() {
 }
 
 #[test]
+#[ignore = "See https://github.com/Z3Prover/z3/issues/5702"]
 fn test_solver_unknown() {
     let _ = env_logger::try_init();
     let mut cfg = Config::new();
@@ -1251,7 +1252,7 @@ fn test_goal_apply_tactic() {
         vec![a.clone(), b.clone()],
     );
 
-    let a_implies_b = ast::Bool::implies(&a, &b);
+    let a_implies_b = ast::Bool::implies(&a, &b).simplify();
     let a_and_a_implies_b = Bool::and(&ctx, &[&a, &a_implies_b]);
 
     let goal = Goal::new(&ctx, false, false, false);

